File: python/25.py
#!/usr/bin/env python

fib_array = [1,1]

while (1):
  next = fib_array[-1] + fib_array[-2]
  if len(str(next)) == 1000:
    print len(fib_array) + 1
    break
  else:
    fib_array.append(next)

File: ruby/25.rb
#!/usr/bin/env ruby

fib_array = [1,1]

begin
  fib = fib_array[-2] + fib_array[-1]
  if fib.to_s.length == 1000
    puts fib_array.length + 1
    exit
  else
    fib_array.push(fib)
  end
end while true
